edb/edgeql/compiler/viewgen.py

```diff
@@ -1502,10 +1502,19 @@ def _normalize_view_ptr_expr(
             )
 
             try:
+                is_linkprop_mutation = (
+                    is_linkprop
+                    and s_ctx.view_rptr is not None
+                    and s_ctx.view_rptr.exprtype.is_mutation()
+                )
+
                 ptrcls = setgen.resolve_ptr(
                     ptrsource,
                     ptrname,
-                    track_ref=False,
+                    track_ref=(
+                        False if not is_linkprop_mutation
+                        else shape_el_desc.ptr_ql
```

Comment:
This seems like it now tracks strictly more. I take it we were actually maybe using the errors sometimes?